unblob/handlers/archive/zip.py

```diff
@@ -147,7 +155,30 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
             file.seek(offset, io.SEEK_SET)
             end_of_central_directory = self.parse_header(file)
 
-            if self.is_zip64_eocd(end_of_central_directory):
+            if end_of_central_directory.offset_of_cd < offset:
+                try:
+                    file.seek(
+                        start_offset + end_of_central_directory.offset_of_cd,
+                        io.SEEK_SET,
+                    )
+                    central_directory = self.cparser_le.cd_file_header_t(file)
+                    if self.is_zip64_cd(central_directory):
+                        file.seek(offset, io.SEEK_SET)
+                        end_of_central_directory = self._parse_zip64(
+                            file, start_offset, offset
+                        )
+                        break
+                except Exception:
+                    logger.info("Malformed zip container, discarding")
```

Comment:
`Exception` is too generic and the try/catch block is too broad. If it's related to seeking out of range, the `if` condition you introduced should be sufficient. If you want to guard it even more you can check that the central directory offset is positive and less than the current EOCD offset.

Since the central directory offset is used twice, you could renamed it this way:

```python
absolute_offset_of_cd = start_offset + end_of_central_directory.offset_of_cd
```

And I would change the condition to:

```
if 0 > absolute_offset_of_cd < offset:
```

Reply:
Nice, changed it according to your suggestion to get ride of `try expect`